### PR TITLE
Fixes #30159 - drop Statistics backend

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Depends on #7857

### CODE REVIEW

File: app/controllers/api/v2/statistics_controller.rb [hunk 1]
```diff
@@ -1,25 +1,21 @@
 module Api
   module V2
+    # This controller just informs about the plugin
     class StatisticsController < V2::BaseController
-      api :GET, "/statistics/", N_("Get statistics")
+      prepend_before_action :fail_and_inform_about_plugin
+
+      resource_description do
+        desc 'This resource has been deprecated, to continue using it please install Foreman Statistics plugin and use its API enpoints.'
```
Comment (ekohl) [inline app/controllers/api/v2/statistics_controller.rb]:
```suggestion
        desc 'This resource has been deprecated, to continue using it please install Foreman Statistics plugin and use its API endpoints.'
```

File: app/controllers/api/v2/statistics_controller.rb [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
+        deprecated true
+      end
 
+      api :GET, "/statistics/", N_("Get statistics")
+      description 'This resource has been deprecated, to continue using it, install Foreman Statistics plugin.'
```
Comment (ekohl) [inline app/controllers/api/v2/statistics_controller.rb]:
```suggestion
      description 'This resource has been deprecated, to continue using it, install the Foreman Statistics plugin.'
```

**Comment by ezr-ondrej** [comment]
Rebased, so the tests should go :green_apple:

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !

**Pull request merged**